test/lightning/runs/prom_ex_plugin_test.exs

```diff
@@ -49,11 +49,10 @@ defmodule Lightning.Runs.PromExPluginText do
           [@stalled_run_threshold_seconds]
         }
 
-      [stalled_run_polling | _] =
-        PromExPlugin.polling_metrics(plugin_config())
+      stalled_run_polling =
+        plugin_config() |> find_metric_group(:lightning_stalled_run_metrics)
```

Comment:
@jyeshe I disagree :). 

While a reviewer will look at this code once, ti will be read many times by:
- people trying to understand the code so that they can change or maintain it
- people who have made a change that has resulted in the test breaking.

For both these groups, I believe that the abstraction produced by helper method, such as the above, allows the reader to compartmentalise parts of the test easily and allows them to slowly immerse themselves in the complexity. With the helper method, the reader can initially step over the complexity and get a feel for the test in general before deciding if they need to delve deeper into how a metric is found.

Also, once they have understood the nature of what find_metric_group provides, it is easy to mentally map it in future tests. Whereas if I with your suggestion, I feel that the reader has to read both lines of code in each  test that they appear before that they can reach the conclusion that it is doing exactly what the previous tests' two lines of code were doing.

Happy to have a higher-bandwidth conversation about this.

Reply:
If it's something that does make sense yet for you, on the idea `find_metric_group` replacement, please consider `Enum.find` in the place of `find_metric`.

Reply:
@jyeshe Thank you for the thoughtful and thought-provoking response and apologies for missing that you did, indeed, mention future readers.

On reflection, it does still make sense for me - so if you are still happy for me to leave it in, so am I.

Also, thank you for the tip re: `Enum.find` I have updated accordingly in a new commit for your perusal.

Reply:
Cool, thank you Rory!

Reply:
The ideal would having `Enum.find(metrics, ...)` instead of `find_metric(metric)` to avoid duplicating the find provided by the language but it's ok.